File: 1678-goal-parser-interpretation/1678-goal-parser-interpretation.java
class Solution {
    public String interpret(String command) {
        char[] arr = command.toCharArray();
        StringBuilder sb = new StringBuilder();
        int i = 0;
        while (i < arr.length) {
            if (arr[i] == 'G') {
                sb.append('G');
                i++;
            } else if (arr[i] == '(' && arr[i + 1] == ')') {
                sb.append('o');
                i += 2;
            }
            else {
                sb.append("al");
                i+=4;
            }
        }
        return sb.toString();
    }
}

File: 1678-goal-parser-interpretation/1678-goal-parser-interpretation.cs
public class Solution {
    public string Interpret(string command)
        {
            string deciphered = "";
            char[] arr = command.ToCharArray();
            for (int i = 0; i < arr.Length; i++)
            {
                if (arr[i] == 'G')
                {
                    deciphered += 'G';
                }
                else if (arr[i] == '(' && arr[i + 1] == ')')
                {
                    deciphered += 'o';
                }
                else if (arr[i] == 'a' && arr[i + 1] == 'l')
                {
                    deciphered += "al";
                }
            }
            Console.WriteLine(deciphered);
            return deciphered;
        }
}